marshaller_test.go

```diff
@@ -1474,5 +1474,147 @@ var _ = Describe("Marshaller", func() {
 				Expect(actualQuery).To(Equal(expectedQuery))
 			})
 		})
+
+		Context("when a struct with where clause with joiner=OR passed", func() {
+			BeforeEach(func() {
+				soqlStruct = orSOQLQuery{
+					WhereClause: positionOrDeptCriteria{
+						Title:      "Purchasing Manager",
+						Department: "Accounting",
+					},
+				}
+				expectedQuery = "SELECT Name,Email,Phone FROM Contact WHERE Title = 'Purchasing Manager' OR Department = 'Accounting'"
+			})
+
+			It("returns properly constructed soql query", func() {
+				Expect(err).ToNot(HaveOccurred())
+				Expect(actualQuery).To(Equal(expectedQuery))
+			})
+		})
+
+		Context("when a struct with where clause with joiner=or passed", func() {
+			BeforeEach(func() {
+				soqlStruct = orLowerSOQLQuery{
+					WhereClause: positionOrDeptCriteria{
+						Title:      "Purchasing Manager",
+						Department: "Accounting",
+					},
+				}
+				expectedQuery = "SELECT Name,Email,Phone FROM Contact WHERE Title = 'Purchasing Manager' OR Department = 'Accounting'"
+			})
+
+			It("returns properly constructed soql query", func() {
+				Expect(err).ToNot(HaveOccurred())
+				Expect(actualQuery).To(Equal(expectedQuery))
+			})
+		})
+
+		Context("when a struct with where clause with joiner=AND passed", func() {
+			BeforeEach(func() {
+				soqlStruct = andSOQLQuery{
+					WhereClause: positionOrDeptCriteria{
+						Title:      "Purchasing Manager",
+						Department: "Accounting",
+					},
+				}
+				expectedQuery = "SELECT Name,Email,Phone FROM Contact WHERE Title = 'Purchasing Manager' AND Department = 'Accounting'"
+			})
+
+			It("returns properly constructed soql query", func() {
+				Expect(err).ToNot(HaveOccurred())
+				Expect(actualQuery).To(Equal(expectedQuery))
+			})
+		})
+
+		Context("when a struct with where clause with joiner=ELSE (an invalid value) passed", func() {
+			BeforeEach(func() {
+				soqlStruct = invalidJoinerSOQLQuery{
+					WhereClause: positionOrDeptCriteria{
+						Title:      "Purchasing Manager",
+						Department: "Accounting",
+					},
+				}
+				expectedQuery = "SELECT Name,Email,Phone FROM Contact WHERE Title = 'Purchasing Manager' AND Department = 'Accounting'"
+			})
+
+			It("returns properly constructed soql query", func() {
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
I think we should return `ErrInvalidTag` in this case. If value of `joiner` is anything other than `or` or `and` (case insensitive) then we should throw an error. It will prevent surprises for the user.